src/main.c

```diff
@@ -505,6 +505,88 @@ np2srv_sm_oper_cb(sr_session_ctx_t *session, uint32_t UNUSED(sub_id), const char
     return rc;
 }
 
+#ifdef NC_ENABLED_SSH_TLS
+
+void
+np2srv_cert_exp_notif_cb(const char *expiration_time, const char *xpath, void *user_data)
+{
+    sr_session_ctx_t *sr_sess = user_data;
+    const struct ly_ctx *ly_ctx = NULL;
+    int rc, stop_thread = 0;
+    struct lyd_node *ntf = NULL;
+
+    ly_ctx = sr_acquire_context(np2srv.sr_conn);
+    if (!ly_ctx) {
+        ERR("Failed to acquire sysrepo context.");
+        stop_thread = 1;
+        goto cleanup;
+    }
+
+    rc = lyd_new_path(NULL, ly_ctx, xpath, expiration_time, 0, &ntf);
+    if (rc) {
+        ERR("Failed to create certificate expiration notification data.");
+        stop_thread = 1;
+        goto cleanup;
+    }
+
+    rc = sr_notif_send_tree(sr_sess, ntf, 0, 0);
+    if (rc) {
+        ERR("Failed to send certificate expiration notification.");
+        stop_thread = 1;
+        goto cleanup;
+    }
+
+cleanup:
+    lyd_free_tree(ntf);
+    if (ly_ctx) {
+        sr_release_context(np2srv.sr_conn);
+    }
+    if (stop_thread) {
+        nc_server_notif_cert_expiration_thread_stop(1);
+    }
+}
+
+static int
+np2srv_start_cert_exp_notif_thread()
```

Comment:
In function declarations we always specify at least `void` if there are no parameters, use empty parentheses only when calling such a function.